raiden/network/proxies/token_network.py

```diff
@@ -809,96 +864,241 @@ def _set_total_deposit(
+                    raise RaidenRecoverableError("Deposit overflow")
+
+                total_deposit_done = our_details.deposit >= total_deposit
+                if total_deposit_done:
+                    raise RaidenRecoverableError("Requested total deposit was already performed")
+
+                token_network_deposit_limit = self.token_network_deposit_limit(
+                    block_identifier=receipt["blockHash"]
+                )
+
+                network_balance = self.token.balance_of(
+                    address=Address(self.address), block_identifier=receipt["blockHash"]
+                )
+
+                if network_balance + deposit_amount > token_network_deposit_limit:
+                    msg = (
+                        f"Deposit of {deposit_amount} would have "
+                        f"exceeded the token network deposit limit."
+                    )
+                    raise RaidenRecoverableError(msg)
+
+                channel_participant_deposit_limit = self.channel_participant_deposit_limit(
+                    block_identifier=receipt["blockHash"]
+                )
+                if total_deposit > channel_participant_deposit_limit:
+                    msg = (
+                        f"Deposit of {total_deposit} is larger than the "
+                        f"channel participant deposit limit"
+                    )
+                    raise RaidenRecoverableError(msg)
+
+                has_sufficient_balance = (
+                    self.token.balance_of(self.node_address, given_block_identifier)
+                    < amount_to_deposit
+                )
+                if not has_sufficient_balance:
+                    raise RaidenRecoverableError(
+                        "The account does not have enough balance to complete the deposit"
+                    )
+
+                allowance = self.token.allowance(
+                    owner=self.node_address,
+                    spender=Address(self.address),
+                    block_identifier=failed_at_blockhash,
+                )
+                if allowance < amount_to_deposit:
+                    msg = (
+                        f"The allowance of the {amount_to_deposit} deposit changed. "
+                        f"Check concurrent deposits "
+                        f"for the same token network but different proxies."
+                    )
+                    raise RaidenUnrecoverableError(msg)
+
+                latest_deposit = self._detail_participant(
+                    channel_identifier=channel_identifier,
+                    detail_for=self.node_address,
+                    partner=partner,
+                    block_identifier=failed_at_blockhash,
+                ).deposit
+                if latest_deposit < total_deposit:
+                    raise RaidenUnrecoverableError("The tokens were not transferred")
+
+                raise RaidenUnrecoverableError("Unlocked failed for an unknown reason")
```

Comment:
Hummm, because we have a require for the token transfer:

https://github.com/raiden-network/raiden-contracts/blob/1db8c337682185a52177bb52bf30cc44735a7a3a/raiden_contracts/data/source/raiden/TokenNetwork.sol#L365

I think the fallback should be recoverable.

Reply:
Done